src/Symfony/Component/Clock/MockClock.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class MockClock implements TimeInterface
+{
+    private \DateTimeImmutable $now;
+
+    public function __construct(\DateTimeImmutable|\DateTimeZone|string $now = null)
```

Comment:
In my experience a test clock implementation benefits from having a `freeze(DateTimeImmutable): void` method. This is more useful for integration tests where you might need to set the clock at a specific time or different times in a row.

Also probably out of scope but I like to pair it up with a PHPUnit extension which unfreeze it between each test (a bit similar to https://github.com/dmaicher/doctrine-test-bundle)

Reply:
A MockClock is always frozen so I'm not sure what these would do. To advance the clock, one can call `$mockClock->sleep(123.456)`, eg between test cases with a phpunit extension, or between the steps of a test case when needed.

Reply:
Ah missed that part: yes I went the other road of having a `FreezableClock`, which is not frozen by default, but can be arbitrarily frozen. So it is probably better to go with a dedicated class instead

Reply:
I have the feeling that it's indeed better to have its own class and keep them immutable.